From bf9ddd04445ec6a3b8c74a8e9105a9de52667212 Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Fri, 23 Feb 2024 14:47:19 +0100
Subject: [PATCH 0626/1695] plat-stm32mp2: scmi_server: remove OSPI2 reset for
 STM32MP23

OSPI2 doesn't exist on STM32MP23x family.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 908c0e9941e9d84f57056faf362179205180b543 plat-stm32mp2: scmi_server: remove OSPI2 reset for STM32MP23]
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Change-Id: I312f5c80ace974952d11072c5cef0464039213b9
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/362570
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/arch/arm/plat-stm32mp2/scmi_server_scpfw.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/core/arch/arm/plat-stm32mp2/scmi_server_scpfw.c b/core/arch/arm/plat-stm32mp2/scmi_server_scpfw.c
index bcaff2a6e..6e48637a3 100644
--- a/core/arch/arm/plat-stm32mp2/scmi_server_scpfw.c
+++ b/core/arch/arm/plat-stm32mp2/scmi_server_scpfw.c
@@ -433,7 +433,9 @@ static struct stm32_scmi_rd stm32_scmi_reset[] = {
 	RESET_CELL(RST_SCMI_OSPI2, OSPI2_R, OSPI2_BASE, "ospi2"),
 #endif
 	RESET_CELL(RST_SCMI_OSPI1DLL, OSPI1DLL_R, OSPI1_BASE, "ospi1_ddl"),
+#ifdef CFG_STM32MP25
 	RESET_CELL(RST_SCMI_OSPI2DLL, OSPI2DLL_R, OSPI2_BASE, "ospi2_ddl"),
+#endif
 };
 
 static const struct channel_resources scmi_channel[] = {
-- 
2.47.2

